admin/schema.php

```diff
@@ -471,7 +473,7 @@
 $g_upgrade[61] = array( 'CreateTableSQL', array( db_get_table( 'tag' ), "
 	id						I		UNSIGNED NOTNULL PRIMARY AUTOINCREMENT,
 	user_id					I		UNSIGNED NOTNULL DEFAULT '0',
-	name					C(100)	NOTNULL PRIMARY DEFAULT \" '' \",
+	name					C(100)	NOTNULL UNIQUE DEFAULT \" '' \",
```

Comment:
You can't just do that.

This will introduce a schema discrepancy between existing installations, which will have the composite (id, name) primary key, while new installations will have 2 indexes primary (id) and unique(name).  Moreover, the the latter will be a duplicate of _idx_tag_name_ created at schema 180 which should probably be dropped.

Additional schema steps must be added to ensure the schema is identical in both scenarios.

Since this composite primary key issue can be considered as a bug independent from the SQLite3 support, a separate Issue should be created to track it and give it visibility in the change log. It can still be fixed as part of this PR if you want, with proper reference in relevant commit(s).

Reply:
I removed UNIQUE.

Checking the use of the ‘tag’ table showed:
- Creating a new tag with manual uniqueness check of the name by tag_is_unique().
- Updating tag data by id with manual uniqueness check by tag_is_unique().
- Deleting a tag by id.
- Selecting a specific tag by id list.
- Selecting a specific tag by name (two functions).
- Selecting all tags.
- Counting all tags.
- Selecting all tags that are not yet assigned to a bug by id.

I don't see any complex queries on the table, and even uniqueness is not checked by database tools. I think that only two indexes on `id` and `name` columns are enough for efficient work, and they are already there.

I tested the scheme also on MySQL 9.0.1 and PostgreSQL 16.1 - everything works perfectly. 😊

And I have a two news for ADODb - bad news and good news. The good news is that everything works, even with UNIQUE. The bad news is that it should NOT work with UNIQUE. Unfortunately the UNIQUE attribute seems to be removed and ignored by ADODb from the SQL query, although it should result in either setting the appropriate table column attribute or creating a column index with the column name, depending on the database type. It's just ignored in all three tested bases. ☹️